src/component/SignupBody.jsx

```diff
@@ -29,31 +28,68 @@ export const SignupBody = ()=>{
             <label htmlFor="Name" className="para4  m-8 label">FirstName</label>
 
                 <input type="text"   id="firstName" className="input border-color-gray margin-none"
-                    placeholder="Enter Your First Name" onChange={(e)=>setFirstName(e.target.value)} required/>
+                    placeholder="Enter Your First Name" onChange={(e)=>{
+                        setError("noerror")
+                        setFirstName(e.target.value)}} 
+                        onFocus={()=>setError("noerror")}
+                        required/>
 
                      <label htmlFor="lastname" className="para4  m-8 label">LastName</label>
 
                 <input type="text" id="lastName" className="input border-color-gray margin-none"
-                    placeholder="Enter Your Last Name" onChange={(e)=>setLastname(e.target.value)} required/>
+                    placeholder="Enter Your Last Name" onChange={(e)=>{
+                        setError("noerror")
+                        setLastname(e.target.value)} } 
+                        onFocus={()=>setError("noerror")}
+                        required/>
 
                 <label htmlFor="username" className="para4  m-8 label">Email Address</label>
 
                 <input type="email" id="email" className="input border-color-gray margin-none"
-                    placeholder="Enter Your Email Id" onChange={(e)=>setEmail(e.target.value)} required/>
+                    placeholder="Enter Your Email Id" onChange={(e)=>{
+                        setError("noerror")
+                        setEmail(e.target.value)}} 
+                        onFocus={()=>setError("noerror")}
+                        required/>
 
                 <label htmlFor="password" className="para4  m-8 label">Password</label>
 
                 <input type="password" id="password" autoComplete="false" className="input border-color-gray margin-none"
-                    placeholder="Enter Your Password"onChange={(e)=>setPassword(e.target.value)} minLength={8}  required/>
+                    placeholder="Enter Your Password"onChange={(e)=>{
+                        setError("noerror")
+                        setPassword(e.target.value)} }minLength={8}  
+                        onFocus={()=>setError("noerror")}
+                        required/>
 
                 <div className="d-flex align-items-center terms">
                     {/* <div><input type="checkbox" name="number" className="input input-checkbox" value="one" id="one"/></div> */}
                     {/* <div><label className="label" htmlFor="one" >I accept all Terms and Conditions</label></div> */}
                 </div>
                 <button type="submit" onClick={(e)=>{
                         e.preventDefault()
-                SignupHandler(firstName,lastname,email,password)}
-                }
+                        var validRegex = /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/;
```

Comment:
no need to add email validation function, just simply add required attribute to email with input type email